app/models/user_course.rb

```diff
@@ -15,5 +15,17 @@ class UserCourse < ApplicationRecord
 
   accepts_nested_attributes_for :user_course_subjects, allow_destroy: true
 
+  after_create :add_user_course_subjects
+
   enum status: {start: 0, inprogress: 1, finished: 2, canceled: 3}
+
+  def add_user_course_subjects
+    return unless trainee?
+
+    co_subs = course.course_subjects
+    co_subs.each do |co_sub|
+      user_course_subjects.create! course_subject_id: co_sub.id,
+                                      start_time: Time.current
```

Comment:
start_time tôi nghĩ là sau này phải xem lại nhé, vì start time của các subject sau tính toán dựa trên subject trước

Reply:
Đây là thiết đặt mặc định, để null thì không được. Truyền một giá trị vào để nó không null thôi. Phần start_time của subject chắc sẽ xử lý sau.